src/test/java/codeu/controller/RegisterServletTest.java

```diff
@@ -4,31 +4,92 @@
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
 
-public class RegisterServletTest
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
+
+public class RegisterServletTest 
 {
-  private RegisterServlet registerServlet;
-  private HttpServletRequest mockRequest;
-  private PrintWriter mockPrintWriter;
-  private HttpServletResponse mockResponse;
+
+	private RegisterServlet registerServlet;
+	private HttpServletRequest mockRequest;
+	private PrintWriter mockPrintWriter;
+	private HttpServletResponse mockResponse;
+
+	@Before
+	public void setup() throws IOException 
+	{
+		registerServlet = new RegisterServlet();
+		mockRequest = Mockito.mock(HttpServletRequest.class);
+		mockPrintWriter = Mockito.mock(PrintWriter.class);
+		mockResponse = Mockito.mock(HttpServletResponse.class);
+		Mockito.when(mockResponse.getWriter()).thenReturn(mockPrintWriter);
+	}//setup
+
+	@Test
+	public void testDoGet() throws IOException, ServletException
+	{
+		registerServlet.doGet(mockRequest, mockResponse);
+		Mockito.verify(mockPrintWriter).println("<h1>RegisterServlet GET request.</h1>");
+	}//testDoGet
  
-  @Before
-  public void setup() throws IOException 
-  {
-    registerServlet = new RegisterServlet();
-    mockRequest = Mockito.mock(HttpServletRequest.class);
-    mockPrintWriter = Mockito.mock(PrintWriter.class);
-    mockResponse = Mockito.mock(HttpServletResponse.class);
-    Mockito.when(mockResponse.getWriter()).thenReturn(mockPrintWriter);
-  }//setup
-
-  @Test
-  public void testDoGet() throws IOException, ServletException
-  {
-    registerServlet.doGet(mockRequest, mockResponse);
-    Mockito.verify(mockPrintWriter).println("<h1>RegisterServlet GET request.</h1>");
-  }//testDoGet
+	@Test
+	public void testDoPost_BadUsername() throws IOException, ServletException {
+		Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
+
+		loginServlet.doPost(mockRequest, mockResponse);
+
+		Mockito.verify(mockRequest).setAttribute("error", "Please enter only letters, numbers, and spaces.");
+		Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
```

Comment:
We should also make sure that the user isn't created. I know from our code its pretty obvious that if we show an error the user wont be created. We should still however test for it, since the test would break if someone were to make an incorrect change in the future.